### PR TITLE
London Class 8 - Hamra Khan - JS3 - Week 1

### PR DESCRIPTION
**Pull request by HamraKhan**
**Volunteers: Are you marking this coursework?** _You can find a guide on how to mark this coursework in `HOW_TO_MARK.md` in the root of this repository_

# Your Details

- Your Name: Hamra Khan
- Your City: London
- Your Slack Name: Hamra Khan

# Homework Details

- Module: JS3
- Week: 1

# Notes

- What did you find easy?

- What did you find hard?

- What do you still not understand?

- Any other notes?

-----
[View rendered extra/1-debugger/task.md](https://github.com/HamraKhan/JavaScript-Core-3-Coursework-Week1-London8/blob/london-class-8/hamra-khan/extra/1-debugger/task.md)
[View rendered mandatory/thinking-like-a-programmer/task.md](https://github.com/HamraKhan/JavaScript-Core-3-Coursework-Week1-London8/blob/london-class-8/hamra-khan/mandatory/thinking-like-a-programmer/task.md)

### CODE REVIEW

File: mandatory/thinking-like-a-programmer/task.md [hunk 1]
```diff
@@ -29,12 +29,46 @@ After you've watched these videos I'd like you to answer these questions
...
 ## 2. When trying to solve a challenge, what should you do first?
 
-<!-- Write your answer here -->
+<!-- Write your answer here 
+
+Have your own framework. 
+
+Understand the problem as opposed to just diving in and trying a few solutions without understanding it properly.
+Draw some type of user interface/diagram, as it helps to visualise what you're working on. Or use the rubber duck approach and 
+talk to it and explain the problem (talking out loud).
```
Comment (mahri09) [inline mandatory/thinking-like-a-programmer/task.md:57]: really liked your option I definitely try on me thanks a lot!

File: mandatory/thinking-like-a-programmer/task.md [hunk 2]
```diff
@@ -29,12 +29,46 @@ After you've watched these videos I'd like you to answer these questions
...
+-->
 
 ## 3. What should you do if you get stuck?
 
-<!-- Write your answer here -->
+<!-- Write your answer here 
+
+Divide the problem into smaller problems. It makes it less intimidating and you feel more likely to complete
+the tasks bit by bit. Break it down like a math problem and take it a step at a time as each step is its own 
+task in itself with its own purpose of solving the product.
```
Comment (mahri09) [inline mandatory/thinking-like-a-programmer/task.md:69]: I am in same opinion :))

**Review by mahri09 — COMMENTED** [review]
I was enjoyed to reading your opinions thank Hamra

File: mandatory/thinking-like-a-programmer/task.md [hunk 3]
```diff
@@ -29,12 +29,46 @@ After you've watched these videos I'd like you to answer these questions
...
-<!-- Write your answer here -->
+<!-- Write your answer here 
+
+Have your own framework. 
+
+Understand the problem as opposed to just diving in and trying a few solutions without understanding it properly.
+Draw some type of user interface/diagram, as it helps to visualise what you're working on. Or use the rubber duck approach and 
+talk to it and explain the problem (talking out loud). 
+
+Think of bugs/errors as a challenge not an annoyance because you can learn from it.
```
Comment (ewintram) [inline mandatory/thinking-like-a-programmer/task.md:59]: love this!

File: mandatory/thinking-like-a-programmer/task.md [hunk 4]
```diff
@@ -29,12 +29,46 @@ After you've watched these videos I'd like you to answer these questions
...
 
-<!-- Write your answer here -->
+<!-- Write your answer here 
+
+Divide the problem into smaller problems. It makes it less intimidating and you feel more likely to complete
+the tasks bit by bit. Break it down like a math problem and take it a step at a time as each step is its own 
+task in itself with its own purpose of solving the product. 
+
+If you get stuck and can't figure it out still, then comment out what you've done so far, so you have it as a
+reference and you can start over but still have those references to refer to later if you need.
```
Comment (ewintram) [inline mandatory/thinking-like-a-programmer/task.md:72]: really great advice for yourself to follow :)

**Pull request closed without merging**